### DISCUSSION TITLE
Can you add a keymap that auto names bookmarks right away?

### DISCUSSION THREAD

**Discussion by martin-braun** [alefragnani/vscode-bookmarks#620] [Q&A] [answered]
I'd like to replicate the Vim behavior in some sense. Each final letter on my keymap should name the bookmark accordingly. I also want to jump to any bookmark in the same fashion. Is this possible with this extension?

Ref: https://github.com/vscode-neovim/vscode-neovim/issues/1223#issuecomment-1616665902

**Comment by alefragnani** [comment, marked as answer]
Hi @martin-braun ,

Bookmarks supports _labels_, and you can play with the `bookmarks.label.suggestion` setting, so the extension will _suggest_ the label for you. But right now, those suggestions are limited to the selected text or the entire line.

Navigation/Jumping, on the other hand, there is not _quick navigation_ concept, as the bookmarks are just positions with (possible) labels to identify it, and you can only navigate to the previous/prior bookmark (line number related) or some random bookmark base on its line content/label. So, a Vim like behavior wouldn't be possible, unfortunately.

I think to have replicate the Vim behavior, a new reengineered base concept would be required. Probably a separate extension for that. 

Hope this helps

**Reply by martin-braun** [reply]
Thanks for the response. I'd like to look at the code to see how I could contribute to enable these motions, but I came to the conclusion that you closed-sourced it, thus I'm not interested anyways.

Nonetheless your respond was useful to understand that this project isn't the right project for the task. Have a nice day.
